File: components/pages/measuring.js
import { html, css } from 'https://cdn.jsdelivr.net/gh/lit/dist@2/core/lit-core.min.js';
import { QuestionElement } from '../base/question-element.js';
import { MEASURING_MODE } from '../../game-pages.js';

class MeasuringMode extends QuestionElement {
  render() {
    return html`
      <div class="game-page">
        <div class="game-header">
          <div class="game-title">${MEASURING_MODE.title}</div>
          <div class="game-info">
            Cost: <span>Draw 3, pick 1</span> Time: <span>5 Minutes</span>
          </div>
        </div>

        <div class="game-question">Compared to me, are you closer to or further from _______?</div>

        ${MEASURING_MODE.categories.map(
          (cat) => html`
            <div class="category">
              <div class="category-title">${cat.title}:</div>
              ${cat.fields.map(
                (item) => html`
                  <persistent-input id="${MEASURING_MODE.title}.${item.id}"
                    >${item.label}</persistent-input
                  >
                `,
              )}
            </div>
          `,
        )}
      </div>
    `;
  }
}

customElements.define('measuring-mode', MeasuringMode);

File: src/pages/measuring.rs
use crate::{components::PersistentInput, constants};
use yew::prelude::*;

#[function_component(Measuring)]
pub fn measuring_mode() -> Html {
    let constants::GameMode { title, categories } = constants::measuring_mode();
    html! {
        <div class="game-page">
            <div class="game-header">
                <div class="game-title">{ &title }</div>
                <div class="game-info">
                    { "Cost: " }
                    <span>{ "Draw 3, Pick 1" }</span>
                    { " Time: " }
                    <span>{ "5 Minutes" }</span>
                </div>
            </div>
            <div class="game-question">
                { "Compared to me, are you closer to or further from _______?" }
            </div>
            { for categories.iter().map(|cat| html! {
                    <div class="category">
                        <div class="category-title">{ &cat.title }{":"}</div>
                        { for cat.fields.iter().map(|item| html! {
                            <PersistentInput
                                id={format!("{}.{}", title, item.id)}
                                label={item.label}
                            />
                        }) }
                    </div>
                }) }
        </div>
    }
}
